src/main/java/org/icgc/argo/program_service/services/InvitationService.java

```diff
@@ -39,17 +49,23 @@ public UUID inviteUser(@NotNull ProgramEntity program,
     return invitation.getId();
   }
 
-  public boolean acceptInvite(UUID invitationId) {
-    val invitation = invitationRepository.findById(invitationId);
-    if (invitation.isPresent()) {
-      val i = invitation.get();
-      i.accept();
-      val email = invitation.get().getUserEmail();
-      egoService.joinProgram(email, i.getProgram().getShortName(), i.getRole());
-      invitationRepository.save(invitation.get());
-      return true;
-    } else {
-      return false;
-    }
+  public EgoUser acceptInvite(UUID invitationId) throws NotFoundException {
+    val invitation = invitationRepository
+            .findById(invitationId)
+            .orElseThrow(() ->
+              new NotFoundException(format("Cannot find invitation with id '%s' ", invitationId)));
+    invitation.accept();
+    egoService.joinProgram(invitation.getUserEmail(), invitation.getProgram().getShortName(), invitation.getRole());
+    invitationRepository.save(invitation);
+    return convertInvitationToEgoUser(invitation);
+  }
+
+  private EgoUser convertInvitationToEgoUser(@NonNull JoinProgramInvite invite){
+    return EgoUser.builder()
+            .email(invite.getUserEmail())
+            .firstName(invite.getFirstName())
+            .lastName(invite.getLastName())
+            .role(invite.getRole())
+            .build();
```

Comment:
please replace this with the ProgramConverter interface method:
```java
  @Mapping(target = "id", ignore = true)
  @Mapping(target = "type", ignore = true)
  @Mapping(target = "status", ignore = true)
  @Mapping(target = "email", source = "userEmail")
  EgoUser joinProgramInviteToEgoUser(JoinProgramInvite invite);
```

which will automatically generate this:
```java

    @Override
    public EgoUser joinProgramInviteToEgoUser(JoinProgramInvite invite) {
        if ( invite == null ) {
            return null;
        }

        EgoUserBuilder egoUser = EgoUser.builder();

        if ( invite.getUserEmail() != null ) {
            egoUser.email( invite.getUserEmail() );
        }
        if ( invite.getFirstName() != null ) {
            egoUser.firstName( invite.getFirstName() );
        }
        if ( invite.getLastName() != null ) {
            egoUser.lastName( invite.getLastName() );
        }
        if ( invite.getRole() != null ) {
            egoUser.role( invite.getRole() );
        }

        return egoUser.build();
    }

```

Again, if you add a new field to EgoUser, mapstruct will catch it at compile time. In the current manual mapping, the system would fail at runtime when it sees that the new field is null